pkg/daemon/ceph/client/osd.go

```diff
@@ -355,3 +355,13 @@ func OSDOkToStop(context *clusterd.Context, clusterInfo *ClusterInfo, osdID, max
 
 	return stats.OSDs, nil
 }
+
+// OsdPrimaryAffinity assign primary-affinity (within range [0.0, 1.0]) to specific OSD.
+func OsdPrimaryAffinity(context *clusterd.Context, clusterInfo *ClusterInfo, osdID int, affinity string) error {
+	args := []string{"osd", "primary-affinity", fmt.Sprintf("osd.%d", osdID), affinity}
```

Comment:
Please 2 log info message:

- One that starts the sequence `setting osd primary affinity...`
- One that reports success: `successfully applied osd primary...`